src/avl.py

```diff
@@ -0,0 +1,468 @@
+import json
+import logging
+
+from hypothesis import given
+import hypothesis.strategies as st
```

Comment:
maybe guard the testing imports so that they only happen if you're actually running a test. better than forcing any user of the lib to load testing stuff